### DISCUSSION TITLE
Prevent lockFileMaintenance from updating direct dependencies

### DISCUSSION THREAD

**Discussion by steverep** [renovatebot/renovate#21604] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have direct dependencies pinned in `package.json`.  When any direct dependency is also a sub-dependency of another package, lock file maintenance has no qualms about updating those direct dependencies with a newer version as a duplicate.  I would like to prevent this but I don't see a way to do so in the configuration.

### What you tried so far.

I do not see a way to accomplish this.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Unfortunately this is perhaps not possible. What you're seeing right now is simply the result of:
- delete lock file 
- run npm install
- commit new lock file

Maybe an npm dedupe command might help, but I'm not sure if one exists which satisfies your example

**Reply by steverep** [reply]
Then, instead would it be possible to either:
1. Add warnings to the PR body citing packages where this occurred, or
2. Add an option to update the affected packages in the same PR

Right now the only way to find occurrences is to manual review and act on the diff, which is of course time consuming and prone to error.

**Reply by viceice** [reply]
no, because we don't analyze the lockfile after generation

**Reply by steverep** [reply]
Is it impossible to do so?

**Reply by viceice** [reply]
currently yes, there are no plans to change that behavior
